examples/experimental/dagster-airlift/dagster_airlift/core/sensor.py

```diff
@@ -157,9 +157,7 @@ def materializations_and_requests_from_batch_iter(
         offset=offset,
     )
     for i, dag_run in enumerate(runs):
-        mats: List[AssetMaterialization] = []
-        mats.extend(materializations_for_dag_run(dag_run, airflow_data))
-        for task_run in airflow_instance.get_task_instance_batch(
+        task_instances = airflow_instance.get_task_instance_batch(
```

Comment:
doesn't this get overidden N times?

Reply:
task instances are scoped to dag run (a task instance is just a run of task), so we need to retrieve them on a per-dag-run basis

Reply:
well that makes perfect sense doesn't it